cedars/app/db.py

```diff
@@ -602,42 +665,37 @@ def get_patient_annotation_ids(p_id, reviewed=False, key="_id"):
 
     return res
 
-
 def get_annotation_date(annotation_id):
     """
     Retrives the event date for an annotation.
     """
     logger.debug(f"Retriving date on annotation #{ObjectId(annotation_id)}.")
     annotation = mongo.db["ANNOTATIONS"].find_one({"_id": ObjectId(annotation_id)})
-    if "event_date" in annotation.keys():
-        return annotation["event_date"]
-    return None
+    patient_id = annotation["patient_id"]
 
+    return get_event_date(patient_id)
 
 def get_event_date(patient_id):
     """
-    Find the event date from the annotations for a patient.
+    Find the event date for a patient.
     """
     logger.debug(f"Retriving event date for patient #{patient_id}.")
-    annotations = mongo.db["ANNOTATIONS"].find({"patient_id": patient_id,
-                                               "event_date": {"$ne": None}}).sort([("event_date", 1)])
-    annotations = list(annotations)
-    if len(annotations) > 0:
-        return annotations[0]["event_date"]
+    patient = mongo.db["PATIENTS"].find_one({"patient_id": patient_id})
 
-    return None
+    return patient['event_date']
 
 
 def get_event_date_sentences(patient_id):
     """
     Find the event date from the annotations for a patient.
     """
     logger.debug(f"Retriving event date for patient #{patient_id}.")
+    event_date = get_event_date(patient_id)
```

Comment:
as you suggested, the patient should have an `event_annotation_id`

`delete_event_date` - should remove this field as well.